system/lib/pthread/library_pthread.c

```diff
@@ -268,7 +268,19 @@ int usleep(unsigned usec)
 	}
 #ifdef __EMSCRIPTEN__
 	emscripten_conditional_set_current_thread_status(EM_THREAD_STATUS_SLEEPING, EM_THREAD_STATUS_RUNNING);
-#endif
+#endif	
+}
+
+int nanosleep(const struct timespec *req, struct timespec *rem)
+{
+	if (!req || req->tv_nsec < 0 || req->tv_nsec > 999999999L || req->tv_sec < 0) return EINVAL;
+	do_sleep(req->tv_sec * 1000.0 + req->tv_nsec / 1e6);
```

Comment:
`tv_nsec` is an integer, and so is `1e6`, so that division will be an integer division and lose precision, but i guess that's ok here? but then why is the former element a float?